### ISSUE TITLE
spinner: does not work fullly with decimal places

### ISSUE THREAD

**Issue by nimo23** [primefaces/primefaces#9516]
### Describe the bug

The `p:spinner` does not work with decimal places **greater than 3**. It stops to increment always after few incrementations (as if there were a `spinner.max `enabled, but it's not.). For example:

```
<p:spinner 
		id="spinner1" 
		value="#{bean.plainDoubleValue}" 
		min="0"
		decimalPlaces="8"
		stepFactor="0.00001000"
		rotate="false">
```

1. increment by the button up
2. it stops after 10 increments

Look at the reproducer!

### Reproducer

[primefaces-test.zip](https://github.com/primefaces/primefaces/files/10240987/primefaces-test.zip)

### PrimeFaces edition

None

### PrimeFaces version

12

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

17

**Comment by melloware** [comment]
Thanks for the reproducer!

Looks like this was fixed in 13.0 with this ticket: https://github.com/primefaces/primefaces/issues/9442

There is a MonkeyPatch provided in that ticket.

**Comment by nimo23** [comment]
@melloware Thanks!
